packages/lib/server/username.ts

```diff
@@ -34,33 +33,23 @@ type CustomNextApiHandler<T = unknown> = (
   res: NextApiResponse<T>
 ) => void | Promise<void>;
 
-async function fetchWordlist() {
-  const UPSATCH_ENV_FOUND = process.env.UPSTASH_REDIS_REST_URL && process.env.UPSTASH_REDIS_REST_TOKEN;
-
-  if (!WORDLIST_PATH || !UPSATCH_ENV_FOUND || !IS_CALCOM) {
-    return {};
-  }
-  const redis = Redis.fromEnv();
-
-  const cachedData = await redis.get("premium-username-word-list");
-
-  if (cachedData) {
-    return JSON.parse(cachedData as string);
-  } else {
-    const response = await fetch(WORDLIST_PATH);
-    const data = await response.json();
-    await redis.set("premium-username-word-list", JSON.stringify(data), {
-      ex: 60 * 60 * 24 * 7,
-    });
-
-    return data;
+export async function isBlacklisted(username: string) {
+  // NodeJS forEach is very, very fast (these days) so even though we only have to construct the Set
+  // once every few iterations, it doesn't add much overhead.
+  if (!cachedData.size && process.env.USERNAME_BLACKLIST_URL) {
+    await fetch(process.env.USERNAME_BLACKLIST_URL).then(async (resp) =>
+      (await resp.text()).split("\n").forEach(cachedData.add, cachedData)
```

Comment:
I contemplated plaintext vs json, but I prefer a plaintext file as that feels more natural; a simple list of usernames without. Needless to say split is also a lil faster than json.parse()

I'm also using then, this prevents having to temporarily assign a huge text file to a variable.